components/script/dom/shadowroot.rs

```diff
@@ -309,6 +283,48 @@ impl ShadowRootMethods for ShadowRoot {
     }
 }
 
+impl VirtualMethods for ShadowRoot {
```

Comment:
Make sure https://github.com/servo/servo/blob/main/components/script/dom/virtualmethods.rs#L158 is updated with the new types as well. It's a footgun with the handwritten inheritance implementation.